include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+     * vector with the correct type deduced by the Extractor you passed as the
+     * @p variable argument.
+     *
+     * Before you can call this method, you need to call the
+     * extract_local_dof_values() method at least once, passing the same
+     * @p global_vector_name string, and the same type for the variable @p dummy.
+     *
+     * If you have not previously called the extract_local_dof_values() method,
+     * this function will throw an exception.
+     *
+     * For this function to work properly, the underlying FEValues,
+     * FEFaceValues, or FESubfaceValues object for which you called one of the
+     * reinit() functions, must have computed the information you are
+     * requesting. To do so, the update_gradients flag must be an element of the
+     * list of UpdateFlags that you passed to the constructor of this object.
+     * See The interplay of UpdateFlags, Mapping, and FiniteElement in FEValues
+     * for more information.
+     */
+    template <typename Extractor, typename Number = double>
+    const std::vector<
+      typename OutputType<dim, spacedim, Number, Extractor>::curl_type> &
+    get_curls(const std::string &global_vector_name,
+              const Extractor &  variable,
+              const Number       dummy = Number(0));
+
+    /**
+     * An entry with this name does not exist in the internal boost::any map.
+     */
+    DeclException1(ExcNameNotFound,
+                   std::string,
+                   << "No entry with the name " << arg1 << " exists.");
+
+    /**
+     * The requested type and the stored type are different.
+     */
+    DeclException3(ExcTypeMismatch,
+                   std::string,
+                   const char *,
+                   const char *,
+                   << "The stored type for entry with name \"" << arg1
+                   << "\" is " << arg2 << " but you requested type " << arg3
+                   << ".");
+
+    /**
+     * Return the local dof indices computed by the last reinit() function.
+     */
+    const std::vector<types::global_dof_index> &
+    get_local_dof_indices() const;
+
+    /**
+     * Return the local dof indices of the neighbor cell computed by the last
+     * reinit_neighbor() function.
+     */
+    const std::vector<types::global_dof_index> &
+    get_neighbor_dof_indices() const;
+
+    /**
+     * Return a reference to the used mapping.
+     */
+    const Mapping<dim, spacedim> &
+    get_mapping() const;
+
+  private:
+    /**
+     * The mapping used by the internal FEValues. Make sure it lives
+     * longer than this class.
+     */
+    SmartPointer<const Mapping<dim, spacedim>> mapping;
+
+    /**
+     * The finite element used by the internal FEValues. Make sure it lives
+     * longer than this class.
+     */
+    SmartPointer<const FiniteElement<dim, spacedim>> fe;
+
+    /**
+     * Quadrature formula used to integrate on the current cell, and on its
+     * neighbor.
+     */
+    Quadrature<dim> cell_quad;
```

Comment:
we generally use the word "quad" for two-dimensional cells. maybe spell it out here (and in the next variable)